Expensify/App
# implement character limit checking for custom pronouns

### Details

### Fixed Issues
$ https://github.com/Expensify/App/issues/6492

### Tests


### QA Steps
1. navigate to profile screen
2. Select chose self select pronoun from drop down
3. Enter more that 50 characters
4. Click save
5. Check for input error state and "Exceeds the max length of 50 characters" error
6. Reduce character to below 50
7. Click save
8. See profile saved toast
9. click away from the profile
10. Click back onto the profile and see the changes have been persisted
11. Increase pronoun to 51 characters and select a pre-set in the drop down.
12. Click save
13. see profile saved toast

Retry the above steps for First and Last name. First and Last name will give an API error toast as well as the contextual input error state. This is not part of the change but there is some inconsistency there.


### Tested On


### Screenshots


#### Web
![image](https://user-images.githubusercontent.com/47184118/147467338-c161e908-ba7c-4fac-9b9c-86d9b3858d08.png)
![image](https://user-images.githubusercontent.com/47184118/147978788-77336a00-e6d9-4864-914c-096ff1e4a18c.png)


#### Mobile Web
![image](https://user-images.githubusercontent.com/47184118/147468817-e74ccd71-4ef4-4e37-9356-f3a8b1641fdd.png)

#### Desktop
![image](https://user-images.githubusercontent.com/47184118/147474306-c5ae0736-7680-49e8-b7e4-4affb9a65300.png)


#### iOS
![image](https://user-images.githubusercontent.com/47184118/147478511-9e48031c-d450-4f13-8026-28064ecb5981.png)


#### Android
![image](https://user-images.githubusercontent.com/47184118/147480617-f35a4d73-e716-4cde-807d-56699308537d.png)
